core/pubkeyConverter/bech32PubkeyConverter.go

```diff
@@ -80,44 +80,72 @@ func (bpc *bech32PubkeyConverter) Decode(humanReadable string) ([]byte, error) {
 }
 
 // Encode converts the provided bytes in a bech32 form
-func (bpc *bech32PubkeyConverter) Encode(pkBytes []byte) string {
+func (bpc *bech32PubkeyConverter) Encode(pkBytes []byte) (string, error) {
 	if len(pkBytes) != bpc.len {
-		bpc.log.Debug("bech32PubkeyConverter.Encode PkBytesLength",
+		return "", fmt.Errorf("%w when encoding address, expected length %d, received %d",
+			ErrWrongSize, bpc.len, len(pkBytes))
+	}
+
+	conv, err := bech32.ConvertBits(pkBytes, bech32Config.fromBits, bech32Config.toBits, bech32Config.pad)
+	if err != nil {
+		return "", fmt.Errorf("%w: %s", ErrConvertBits, err.Error())
+	}
+
+	converted, err := bech32.Encode(bpc.prefix, conv)
+	if err != nil {
+		return "", fmt.Errorf("%w: %s", ErrBech32ConvertError, err.Error())
+	}
+
+	return converted, nil
+}
+
+// SilentEncode converts the provided bytes in a bech32 form without returning any error
+func (bpc *bech32PubkeyConverter) SilentEncode(pkBytes []byte, log core.Logger) string {
+	if len(pkBytes) != bpc.len {
```

Comment:
`Encode` method can be reused here thus reducing this method to a call to Encode + a log.Warn in case of error